code/game/gamemodes/miniantags/revenant/revenant_abilities.dm

```diff
@@ -271,50 +271,135 @@
+	var/cast_amount = 50 //How much essence it costs to use
+
+/obj/effect/proc_holder/spell/targeted/revenant/New()
+	..()
+	if(locked)
+		name = "[initial(name)] ([unlock_amount]E)"
+	else
+		name = "[initial(name)] ([cast_amount]E)"
+
+/obj/effect/proc_holder/spell/targeted/revenant/can_cast(mob/living/simple_animal/revenant/user = usr)
+	if(charge_counter < charge_max)
+		return FALSE
+	if(!istype(user)) //Badmins, no. Badmins, don't do it.
+		return TRUE
+	if(user.inhibited)
+		return FALSE
+	if(locked)
+		if(user.essence <= unlock_amount)
+			return FALSE
+	if(user.essence <= cast_amount)
+		return FALSE
+	return TRUE
+
+/obj/effect/proc_holder/spell/targeted/revenant/proc/attempt_cast(mob/living/simple_animal/revenant/user = usr)
+	if(!istype(user)) //If you're not a revenant, it works. Please, please, please don't give this to a non-revenant.
+		name = "[initial(name)]"
+		if(locked)
+			panel = "Revenant Abilities"
+			locked = FALSE
+		return TRUE
+	if(locked)
+		if(!user.castcheck(-unlock_amount))
+			charge_counter = charge_max
+			return FALSE
+		name = "[initial(name)] ([cast_amount]E)"
+		to_chat(user, "<span class='revennotice'>You have unlocked [initial(name)]!</span>")
+		panel = "Revenant Abilities"
+		locked = FALSE
+		charge_counter = charge_max
+		return FALSE
+	if(!user.castcheck(-cast_amount))
+		charge_counter = charge_max
+		return FALSE
+	name = "[initial(name)] ([cast_amount]E)"
+	user.reveal(reveal)
+	user.stun(stun)
+	if(action)
+		action.UpdateButtonIcon()
+	return TRUE
 
-//A note to future coders: do not replace this with an EMP because it will wreck malf AIs and gang dominators and everyone will hate you.
-/obj/effect/proc_holder/spell/aoe_turf/revenant/malfunction/cast(list/targets, mob/living/simple_animal/revenant/user = usr)
+//Induce Madness: Drive someone boooooonkers!
+/obj/effect/proc_holder/spell/targeted/revenant/madness
+	name = "Induce Madness"
+	desc = "Slowly descends someone into madness, causing them to do unpredictable things and, as a side effect, melts their brain."
+	charge_max = 150
+	range = 7
+	include_user = 0
+	stun = 20
+	reveal = 80
+	unlock_amount = 20
+	cast_amount = 30
+	action_icon_state = "induce_madness"
+
+/obj/effect/proc_holder/spell/targeted/revenant/madness/cast(list/targets, mob/living/simple_animal/revenant/user = usr)
 	if(attempt_cast(user))
-		for(var/turf/T in targets)
-			INVOKE_ASYNC(src, .proc/malfunction, T, user)
+		for(var/mob/living/M in targets)
+			if(M.stat)
+				to_chat(user, "<span class='revennotice'>Not enough brain activity for our powers!</span>")
+				return FALSE
+			if(!istype(M, /mob/living/carbon/human))
+				to_chat(user, "<span class='revennotice'>Their brain is too basic for our powers!</span>")
+				return FALSE
+			to_chat(user, "<span class='revenwarning'>We have seeded madness in [M]'s mind! It will continue to fester...</span>")
+			to_chat(M, "<span class='warning'>A horrible feeling decends upon you as your mind goes fuzzy...")
```

Comment:
Am I missing something or does this spell do nothing except send scary messages?

Reply:
i forgot to commit it one sec. it adds an abductor message and gives traumas